STYLEGUIDE.md

```diff
@@ -1,3 +1,40 @@
+# ArangoDB Clang-Format
+
+ArangoDB uses `clang-format` to lint the codebase of [arangodb/arangodb](https://github.com/arangodb/arangodb) & [arangodb/enterprise](https://github.com/arangodb/enterprise).
+
+You are expected to format your local changes via clang-format. Here are your options:
+
+1. Lint with a **pre-commit git hook** (OS dependant)
+    * pre-requisites
+        * Docker
+        * Git 2.9+
+    * how-to (one-time setup)
+        * `cd arangodb`
+        * `cp .githooks/pre-commit-[linux, macos, windows] .git/hooks/pre-commit`
+        * `git config core.hooksPath .git/hooks`
+        * `cd enterprise`
+        * `cp .githooks/pre-commit-[linux, macos, windows] .git/hooks/pre-commit`
+        * `git config core.hooksPath .git/hooks`
+2. Lint with a **local shell script** (OS dependant)
+    * pre-requisites
+        * Docker
+        * Your local changes have been added with `git add`
+    * how-to (must run manually)
+        * `cd arangodb`
+        * `./scripts/clang-format-[linux, macos, windows].sh`
+
+3. Lint with a **codebase-wide format script** (not recommended unless absolutely necessary)
+    * pre-requisites
+        * Docker
+    * how-to
+        * See [pull/15417#comment](https://github.com/arangodb/arangodb/pull/15417#issuecomment-1001211838) 
```

Comment:
I don't think referencing a comment in already closed PR makes sense (e.g. link might become broken), let's provide a guidance in place if necessary.